### PR TITLE
fix: remove nested messages for sort

### LINKED ISSUE #87: Create global order direction to avoid issue in types

### LINKED ISSUE #248: Remove nested messages
When proto are converted to another language, nested messages are in a namespace which complicate a lot the usage of types.

Also, there is some code redundancy that can be removed by removing these messages

### PR DESCRIPTION
**Pull request by esoubiran-aneo**
fix #248
fix #87 

Pour le moment, on ne considère que l'utilisateur ne filtre que sur 1 champ. On verra l'implémentation des filtres poussés dans une prochaine PR.

### CODE REVIEW

**Review by aneojgurhem — COMMENTED** [review]
LGTM but breaking change. Unfortunately, we need this for new filter system. @lemaitre-aneo ?

**Comment by esoubiran-aneo** [comment]
> LGTM but breaking change. Unfortunately, we need this for new filter system. @lemaitre-aneo ?

Breaking change at the type level. Behavior will be the same!

File: Protos/V1/tasks_common.proto [hunk 1]
```diff
@@ -76,6 +77,22 @@ message TaskSummary {
...
+  TASK_SUMMARY_FIELD_TASK_ID = 0; /** The task ID. */
+  TASK_SUMMARY_FIELD_SESSION_ID = 1; /** The session ID. */
+  TASK_SUMMARY_FIELD_STATUS = 2; /** The task status. */
+  TASK_SUMMARY_FIELD_CREATED_AT = 3; /** The task creation date. */
+  TASK_SUMMARY_FIELD_STARTED_AT = 4; /** The task start date. */
+  TASK_SUMMARY_FIELD_ENDED_AT = 5; /** The task end date. */
+  TASK_SUMMARY_FIELD_CREATION_TO_END_DURATION = 6; /** The task duration. Between the creation date and the end date. */
+  TASK_SUMMARY_FIELD_PROCESSING_TO_END_DURATION = 7; /** The task calculated duration. Between the start date and the end date. */
+  TASK_SUMMARY_FIELD_ERROR = 8; /** The error message. Only set if task have failed. */
+}
```
Comment (lemaitre-aneo) [inline Protos/V1/tasks_common.proto:94]: I think we would also need a `TaskField` that would most likely be a `oneof` with `TaskSummaryField`, `TaskOptionFiled`, `TaskOptionGenericField`.

**Review by lemaitre-aneo — COMMENTED** [review]
You can remove the TODO comments, and add a task in the other PR (https://github.com/aneoconsulting/ArmoniK.Api/pull/228)

File: Protos/V1/tasks_common.proto (hunk 1 above)
Comment (esoubiran-aneo) [reply Protos/V1/tasks_common.proto:94]: On va attendre de supprimer le concept de task summary et je continue ensuite de bosser sur ça

Comment (esoubiran-aneo) [reply Protos/V1/tasks_common.proto:94]: en attente de #255

Comment (esoubiran-aneo) [reply Protos/V1/tasks_common.proto:111]: On ne supprime pas le concepte et je retourne bosser.

**Review by esoubiran-aneo — COMMENTED** [review]
@esoubiran-aneo

File: Protos/V1/partitions_common.proto [hunk 2]
```diff
@@ -19,6 +21,25 @@ message PartitionRaw {
   int64 priority = 6; /** The priority of the partition. */
 }
 
+/**
+ * Represents every available field in a partition.
+ */
+enum PartitionRawField {
+  PARTITION_FIELD_UNSPECIFIED = 0; /** Unspecified. */
+  PARTITION_FIELD_ID = 1; /** The partition ID. */
+  PARTITION_FIELD_PARENT_PARTITION_ID = 2; /** The parent partition ID. */
```
Comment (aneojgurhem) [inline Protos/V1/partitions_common.proto:30]: It was plural (IDS)

File: Protos/V1/results_common.proto [hunk 3]
```diff
@@ -23,6 +24,21 @@ message ResultRaw {
   string result_id = 8; /** The result ID. Uniquely generated by the server. */
 }
 
+// TODO: Implements new filters using these fields.
+/**
+ * Represents every available field in a result.
+ */
+enum ResultField {
+  RESULT_FIELD_SESSION_ID = 0; /** The session ID. */
```
Comment (aneojgurhem) [inline Protos/V1/results_common.proto:32]: Unspecified ?

File: Protos/V1/results_common.proto [hunk 4]
```diff
@@ -23,6 +24,21 @@ message ResultRaw {
...
+  RESULT_FIELD_NAME = 1; /** The result name. */
+  RESULT_FIELD_OWNER_TASK_ID = 2; /** The owner task ID. */
+  RESULT_FIELD_STATUS = 3; /** The result status. */
+  RESULT_FIELD_CREATED_AT = 4; /** The result creation date. */
+  RESULT_FIELD_COMPLETED_AT = 5; /** The result completion date. */
+  RESULT_FIELD_WAIT_FOR_DATA_AT = 6; /** The result passing to status wait_for_data date. */
```
Comment (aneojgurhem) [inline Protos/V1/results_common.proto:38]: should not be there. We also need to remove `wait_for_data_at `

File: Protos/V1/tasks_common.proto [hunk 5]
```diff
@@ -93,6 +94,56 @@ message TaskSummary {
...
+  TASK_SUMMARY_FIELD_CREATED_AT = 3; /** The task creation date. */
+  TASK_SUMMARY_FIELD_SUBMITTED_AT = 11; /** The task submission date. */
+  TASK_SUMMARY_FIELD_STARTED_AT = 4; /** The task start date. */
+  TASK_SUMMARY_FIELD_ENDED_AT = 5; /** The task end date. */
+  TASK_SUMMARY_FIELD_CREATION_TO_END_DURATION = 6; /** The task duration. Between the creation date and the end date. */
+  TASK_SUMMARY_FIELD_PROCESSING_TO_END_DURATION = 7; /** The task calculated duration. Between the start date and the end date. */
+  TASK_SUMMARY_FIELD_POD_TTL = 12; /** The pod TTL (Time To Live). */
+  TASK_SUMMARY_POD_HOSTNAME = 13; /** The hostname of the container running the task. */
+  TASK_SUMMARY_RECEIVED_AT = 14; /** When the task is received by the agent. */
+  TASK_SUMMARY_ACQUIRED_AT = 15; /** When the task is acquired by the agent. */
```
Comment (aneojgurhem) [inline Protos/V1/tasks_common.proto:115]:
```suggestion
  TASK_SUMMARY_FIELD_POD_HOSTNAME = 13; /** The hostname of the container running the task. */
  TASK_SUMMARY_FIELD_RECEIVED_AT = 14; /** When the task is received by the agent. */
  TASK_SUMMARY_FIELD_ACQUIRED_AT = 15; /** When the task is acquired by the agent. */
```

File: Protos/V1/applications_common.proto [hunk 6]
```diff
@@ -17,11 +18,26 @@ option csharp_namespace = "ArmoniK.Api.gRPC.V1.Applications";
...
 }
 
+/**
+ * Represents every available field in an application.
+ */
+enum ApplicationRawField {
+  APPLICATION_FIELD_NAME = 0; /** Application name. */
+  APPLICATION_FIELD_VERSION = 1; /** Application version. */
+  APPLICATION_FIELD_NAMESPACE = 2; /** Application namespace. */
+  APPLICATION_FIELD_SERVICE = 3; /** Application service. */
```
Comment (aneojgurhem) [inline Protos/V1/applications_common.proto:32]:
```suggestion
  APPLICATION_RAW_FIELD_NAME = 0; /** Application name. */
  APPLICATION_RAW_FIELD_VERSION = 1; /** Application version. */
  APPLICATION_RAW_FIELD_NAMESPACE = 2; /** Application namespace. */
  APPLICATION_RAW_FIELD_SERVICE = 3; /** Application service. */
```

File: Protos/V1/partitions_common.proto (hunk 2 above)
Comment (aneojgurhem) [reply Protos/V1/partitions_common.proto:30]: And they should be called `PARTITION_RAW_FIELD_*` or the enum should be `PartitionField `

File: Protos/V1/applications_common.proto [hunk 7]
```diff
@@ -17,11 +18,26 @@ option csharp_namespace = "ArmoniK.Api.gRPC.V1.Applications";
...
-
   string namespace = 3; /** Application namespace used in the excecuted class. */
   string service = 4; /** Application service used in the excecuted class. */
 }
 
+/**
+ * Represents every available field in an application.
+ */
+enum ApplicationRawField {
+  APPLICATION_RAW_FIELD_NAME = 0; /** Application name. */
```
Comment (aneojgurhem) [inline Protos/V1/applications_common.proto:29]: Do we put an unspecified in each for coherence sake ?

File: Protos/V1/tasks_common.proto [hunk 8]
```diff
@@ -93,6 +94,58 @@ message TaskSummary {
...
+  TASK_OPTION_FIELD_APPLICATION_NAMESPACE = 6;
+  TASK_OPTION_FIELD_APPLICATION_SERVICE = 7;
+  TASK_OPTION_FIELD_ENGINE_TYPE = 8;
+}
+
+/**
+ * Represents a generic field in a task option.
+ */
+message TaskOptionGenericField {
+  map<string, string> fields = 1; /** The generic fields. */
```
Comment (aneojgurhem) [inline Protos/V1/tasks_common.proto:138]: We want to order by only one field in the TaskData case so it should only be two strings

Comment (aneojgurhem) [reply Protos/V1/tasks_common.proto:138]: Actually, it is only one string : the key. That looks like the thing for the filter not the orderby

File: Protos/V1/applications_common.proto (hunk 7 above)
Comment (esoubiran-aneo) [reply Protos/V1/applications_common.proto:29]: Yes, on peut faire ça. J'ajoute

File: Protos/V1/tasks_common.proto (hunk 8 above)
Comment (esoubiran-aneo) [reply Protos/V1/tasks_common.proto:138]: Ouais, je me suis perdu en cours de route. une map ne semble pas du tout approprié. Je change ça part une simple string !

File: Protos/V1/tasks_common.proto [hunk 9]
```diff
@@ -93,6 +94,60 @@ message TaskSummary {
   google.protobuf.Timestamp acquired_at = 19; /** When the task is acquired by the agent. */
 }
 
+/**
+ * Represents every available field in a task summary.
+ */
+enum TaskSummaryField {
+  TASK_SUMMARY_FIELD_UNSPECIFIED = 16; /** Unspecified */
+  TASK_SUMMARY_FIELD_TASK_ID = 0; /** The task ID. */
```
Comment (aneojgurhem) [inline Protos/V1/tasks_common.proto:102]:
```suggestion
  TASK_SUMMARY_FIELD_UNSPECIFIED = 0; /** Unspecified */
  TASK_SUMMARY_FIELD_TASK_ID = 16; /** The task ID. */
```

Comment (Barbapapazes) [reply Protos/V1/tasks_common.proto:102]: Yes c'est ok

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**